azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -327,6 +350,30 @@ protected void handleException(final Exception exception) throws MojoExecutionEx
         }
     }
 
+    private boolean isFirstRun(Properties prop) throws IOException {

```

Comment:
Should we capture exceptions from this method to avoid blocking maven plugin execution?
Same concern applies to `updateConfigurationFile`.